### PR TITLE
SPOI-5113: Added validation for root operator to be input operator in logical plan validation

### PR DESCRIPTION
**Pull request by ishark**
Added validation for root operator should be input operator in logical plan validation
And updated logical plan test cases to pass this validation

### CODE REVIEW

**Comment by tweise** [comment]
Please use smart commit compliant messages.

File: engine/src/test/resources/dt-site.xml [hunk 1]
```diff
@@ -105,8 +105,8 @@
   <!-- Operator 6, single operator w/o links -->
 
   <property>
-    <name>dt.operator.operator6.classname</name>
-    <value>com.datatorrent.stram.engine.GenericTestOperator</value>
+    <name>dt.operator.operator6.classname</name>
```
Comment (tweise) [inline engine/src/test/resources/dt-site.xml]: Trailing white space - fix IDE settings.

File: engine/src/test/java/com/datatorrent/stram/plan/LogicalPlanTest.java [hunk 2]
```diff
@@ -16,6 +16,7 @@
 package com.datatorrent.stram.plan;
 
 import com.datatorrent.common.util.BaseOperator;
+
```
Comment (tweise) [inline engine/src/test/java/com/datatorrent/stram/plan/LogicalPlanTest.java]: Why are there blank lines (randomly?) added and removed in import blocks?

File: engine/src/test/java/com/datatorrent/stram/engine/TestNonOptionalOutportInputOperator.java [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+
+import com.datatorrent.api.DefaultOutputPort;
+import com.datatorrent.api.annotation.OutputPortFieldAnnotation;
+
+public class TestNonOptionalOutportInputOperator extends TestGeneratorInputOperator{
+
+  @OutputPortFieldAnnotation(optional = false)
+  public final transient DefaultOutputPort<Object> outport1 = new DefaultOutputPort<Object>();
+
+}
```
Comment (tweise) [inline engine/src/test/java/com/datatorrent/stram/engine/TestNonOptionalOutportInputOperator.java]: Add line break.

**Comment by tweise** [comment]
I'm getting a test failure:

mvn test -Dtest=LogicalPlanConfigurationTest

```
2015-07-01 21:54:48,698 [main] DEBUG logical.LogicalPlan <init> - Initializing operator2 as com.datatorrent.stram.engine.GenericTestOperator
Tests run: 17, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 3.702 sec <<< FAILURE!
testLoadFromConfigXml(com.datatorrent.stram.plan.LogicalPlanConfigurationTest)  Time elapsed: 0.139 sec  <<< ERROR!
java.lang.NullPointerException: null
	at com.datatorrent.stram.plan.logical.LogicalPlanConfiguration.populateDAG(LogicalPlanConfiguration.java:1139)
	at com.datatorrent.stram.plan.LogicalPlanConfigurationTest.testLoadFromConfigXml(LogicalPlanConfigurationTest.java:81)


Results :

Tests in error: 
  LogicalPlanConfigurationTest.testLoadFromConfigXml:81 » NullPointer

Tests run: 17, Failures: 0, Errors: 1, Skipped: 0
```

**Comment by ishark** [comment]
Had missed output port change in dt-site.xml on new branch :( Pushed the change in.

**Comment by tweise** [comment]
squash please

**Pull request merged**